jupyterlab_gcpscheduler/src/cron.ts

```diff
@@ -0,0 +1,440 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const DAY_OF_WEEK_NAME_TO_VALUE = {
```

Comment:
Wow! I was hoping there was a library we could use for this. Is this all homegrown code? If so, I'd suggest trying to put this into its own package and publish it to NPM because it seems like a useful bit of functionality that someone else could benefit from.

Maybe even try submitting it to something like https://github.com/harrisiirak/cron-parser since it doesn't appear to have this kind of behavior.

Reply:
Also, it looks like you still may have some snapshot files that need to be updated. The [presubmit](https://github.com/GoogleCloudPlatform/jupyter-extensions/pull/374/checks?check_run_id=1372256442) failed

Reply:
Yeah I ended up writing this because the npm packages I found either had outdated dependencies/no real maintainers. Also since the mocks had different language for the human readable cron, and we need the same for our pantheon version, it felt like something we could commonize there as well. 

The presubmit failures is because of a timezone mis-match, the new commit should fix that.

Reply:
I can work out moving it out to a npm package.

Reply:
I wouldn't move it out for this change. I just thought it might be nice for you to package it independently (if you are interested in long-term support that is!) because you did all of the work to put it together and it seems generally useful.

Reply:
Oh definitely! It would also clean up the code a bit. Also it would be cool to have my own npm package :)